StitchCoreServicesMongoDbRemote/Sources/StitchCoreServicesMongoDbRemote/RemoteInsertManyResult.swift

```diff
@@ -0,0 +1,31 @@
+import Foundation
+import MongoSwift
+
+/// The result of an `insertMany` command on a `RemoteMongoCollection`.
+public struct RemoteInsertManyResult: Decodable {
+    /// Map of the index of the inserted document to the id of the inserted document.
+    public let insertedIds: [Int64: BsonValue]
+    
+    /// Given an ordered array of insertedIds, creates a corresponding `RemoteInsertManyResult`.
+    internal init(fromArray arr: [BsonValue]) {
+        var inserted = [Int64: BsonValue]()
+        for (i, id) in arr.enumerated() {
```

Comment:
[style] Would be nice to see something a bit more functional here:
```swift
zip(arr.indices, arr).forEach { (index, value) in
    inserted[Int64(index)] = value
}
```

Reply:
I can do this, but again, this was code copied from the Swift driver. I guess since this is internal I can change the implementation to be more functional.